contracts/0.4.24/oracle/LidoOracle.sol

```diff
@@ -81,9 +81,37 @@ contract LidoOracle is ILidoOracle, AragonApp {
     bytes32 internal constant MIN_REPORTABLE_EPOCH_ID_POSITION = keccak256("lido.LidoOracle.minReportableEpochId");
     /// @dev the max id of reported epochs
     bytes32 internal constant MAX_REPORTED_EPOCH_ID_POSITION = keccak256("lido.LidoOracle.maxReportedEpochId");
+    /// @dev storage for the last completed report and its time
+    bytes32 internal constant LAST_COMPLETED_REPORT = keccak256("lido.LidoOracle.lastCompletedReport");  // why do we need storage? why not just store `Report public lastReport;` attribute?
```

Comment:
> why do we need storage? why not just store `Report public lastReport;` attribute?
This is [Aragon's unstructured storage](https://github.com/aragon/aragonOS/blob/master/contracts/common/UnstructuredStorage.sol). It protects variables from accidentally changing their slot in the new version of the contract (e.g. the developer added some new variable not to the end of existing variables).